rocky/reports/report_types/definitions.py

```diff
@@ -107,6 +107,34 @@ def to_hostnames(self, input_oois: Iterable[str], valid_time: datetime) -> dict[
 
         return hostnames_by_input_ooi
 
+    def to_human_readable_hostnames(
+        self,
+        input_oois: Iterable[str],
+        valid_time: datetime,
+        extra_indicator: str = ", ...",
+        open_indicator: str = "(",
+        close_indicator: str = ")",
+        hostnames_by_input_ooi: dict | None = None,
```

Comment:
These optional arguments aren't used by any caller. It makes the code more complex and harder to understand than is necessary. In my opinion it is always good to follow the YAGNI principle and not write code that doesn't have a direct use but might be used in the future.

```suggestion
```

Reply:
While I agree with the principles of YAGNI, I believe this is intended to be used as a helper/ utility function. I've seen codebases that provide such utilities, but don't actually use them theirselves or only reference them in their tests. For me it's a gray area so either way is fine, but leaving it out is probably more preferable. I'll leave it up to you.

Imho I'm more concerned about the `hostnames_by_input_ooi` being an optional argument, as it results in a database query as a side effect when the argument is not provided. It would be better to instantiate this variable before passing it to the function, following the principle of dependency injection. This approach would make the function’s behavior more predictable, easier to understand and test.

Reply:
> I believe this is intended to be used as a helper/ utility function.

This is indeed the case here.

> It would be better to instantiate this variable before passing it to the function, following the principle of dependency injection.

I considered this, It would result in extra function calls outside the helper and thus a less easy to use helper. Logically this functionality felt at place next to `to_hostnames` and `to_ips`. However, if the `to_hostnames` is already called close to this proposed function it would result in unnecessary database queries, which I why I ended up with this hybrid solution. Without the call to `to_hostnames` it would become a static method of the `Report` class. Is that the intended solution, or would it have to move to `rocky/reports/report_types/helpers.py` then? (I tried that first, but it felt illogical in practice.) I can move it around either way, let me know what you prefer

Reply:
> I considered this, It would result in extra function calls outside the helper and thus a less easy to use helper. Logically this functionality felt at place next to `to_hostnames` and `to_ips`. However, if the `to_hostnames` is already called close to this proposed function it would result in unnecessary database queries, which I why I ended up with this hybrid solution. Without the call to `to_hostnames` it would become a static method of the `Report` class. Is that the intended solution, or would it have to move to `rocky/reports/report_types/helpers.py` then? (I tried that first, but it felt illogical in practice.) I can move it around either way, let me know what you prefer

Thanks for your reaction. I think it's preferable if we move it to `helpers` module. (Imho the `to_hostnames` and `to_ips` should've also been helper functions outside the `Report` class, since otherwise the class tends to become a god class violating the single responsibility principle; but that's out of scope/ can be refactored soon).

Reply:
It looks like it is not easily possible to move it to the `helpers` module, as that will result in circular imports. I made it a static method for now, it can then be refactored together with the other methods of the `Report` class.